### ISSUE TITLE
Query/Control hangs when terraform_resource refers data from JSON file

### ISSUE THREAD

**Issue by rajlearner17** [turbot/steampipe-plugin-terraform#46]
**Describe the bug**

This is in the case when the  Terraform file is referring to a data source, e.g. defining the policies in the `aws_iam_policy_document`, and the same is referred to in provisioning the `aws_ecr_repository_policy.` 

Please look at the **Note** section below for other observations.

Control hangs 
```
➜ steampipe check control.ecr_repository_policy_prohibit_public_access
⠴ Running 1 control. (0 complete, 1 running, 0 pending, 0 errors)
⠙ Running 1 control. (0 complete, 1 running, 0 pending, 0 errors)
⠴ Running 1 control. (0 complete, 1 running, 0 pending, 0 errors)
⠴ Running 1 control. (0 complete, 1 running, 0 pending, 0 errors)
CTRL + C
+ ECR repository policy should prohibit public access .......... 1 / 1 [          ]
  
  ERROR: execution cancelled
```

**Steampipe version (`steampipe -v`)**
Example: **Steampipe v0.20.10**

**Plugin version (`steampipe plugin list`)**
Example: **0.7.0   | terraform**  

**To reproduce**
Steps to reproduce the behaviour (please include relevant code and/or commands).

Create a TF file with the below.

```
### This works as no data source is provided to associate the policy (inbuilt)

resource "aws_ecr_repository_policy" "foobarworks" {
  repository = "foobar"

  policy = jsonencode(
    {
      Version = "2008-10-17",
      Statement = [
        {
          Effect    = "Allow",
          Principal = "*",
          Action = [
            "ecr:BatchGetImage",
            "ecr:BatchCheckLayerAvailability",
            "ecr:DescribeImages",
            "ecr:DescribeRepositories",
            "ecr:GetDownloadUrlForLayer",
            "ecr:ListImages"
          ],
          Condition = {
            "StringEquals" = {
              "aws:PrincipalOrgID" = "something"
            }
          }
        }
      ]
    }
  )
}

### CHECKING POLICY AS DATA INPUT (Fails when referring to the data source)
resource "aws_ecr_repository" "foo" {
  name = "bar"
}

data "aws_iam_policy_document" "foopolicy" {
  statement {
    sid    = "new policy"
    effect = "Allow"

    principals {
      type        = "AWS"
      identifiers = ["123456789012"]
    }

    actions = [
      "ecr:GetDownloadUrlForLayer",
      "ecr:BatchGetImage",
      "ecr:BatchCheckLayerAvailability",
      "ecr:PutImage",
      "ecr:InitiateLayerUpload",
      "ecr:UploadLayerPart",
      "ecr:CompleteLayerUpload",
      "ecr:DescribeRepositories",
      "ecr:GetRepositoryPolicy",
      "ecr:ListImages",
      "ecr:DeleteRepository",
      "ecr:BatchDeleteImage",
      "ecr:SetRepositoryPolicy",
      "ecr:DeleteRepositoryPolicy",
    ]
  }
}

resource "aws_ecr_repository_policy" "foopolicy" {
  repository = aws_ecr_repository.foo.name
  policy     = data.aws_iam_policy_document.foopolicy.json
}
```

The query to execute 

``` sql
with policy_statement as (
select
  distinct (type || ' ' || name ) as name
from
  terraform_resource ,
  jsonb_array_elements(
    case when ((arguments ->> 'policy') = '')
      then null
      else ((arguments ->> 'policy')::jsonb -> 'Statement') end
) as s
where
  type = 'aws_ecr_repository_policy'
  and (
    (s ->> 'Principal' = '*')
    and ((s ->> 'Condition') is null)
   )
)
select
  type || ' ' || r.name as resource,
  case
    when (arguments ->> 'policy') = ''  then 'ok'
    when s.name is null then 'ok'
    else 'alarm'
  end status,
  case
     when (arguments ->> 'policy') = '' then ' no policy defined'
    when s.name is null then ' not public'
    else ' public'
  end || '.' reason
from
  terraform_resource as r
  left join policy_statement as s on s.name = concat(r.type || ' ' || r.name)
where
  type = 'aws_ecr_repository_policy'
```

**Note-** 

1. Interesting thing is when I execute `.cache off` &` .cache clear`, the query renders as expected 
2. For the first time, the query works (not the control), but when we re-run it, it hangs. 
3. Now same way if the control is hanging for this case, it halts the execution of entire controls across the services
4. Control & query with CTE block mostly hangs
5. It may be very much related to cache, as it works when we execute the query with `.cache off`. The moment the cache is on it hangs

Error from plugin log
```
2023-08-22 10:39:38.535 UTC [TRACE] steampipe-plugin-terraform.plugin: [TRACE] 1692700778380: IndexItem) SatisfiesRequest: satisfiedColumns true satisfiesLimit true satisfiesQuals true
2023-08-22 10:39:38.535 UTC [TRACE] steampipe-plugin-terraform.plugin: [TRACE] 1692700778380: found pending index item to satisfy columns for_each,depends_on,lifecycle,start_line,source,arguments,count,count_src,path,end_line,name,type,provider,_ctx, limit -1, quals: NONE (terraform-1692700778380)
2023-08-22 10:39:38.535 UTC [TRACE] steampipe-plugin-terraform.plugin: [TRACE] 1692700778380: getPendingResultItem returning &{0xc0006de8a0 <nil> terraform-1692700646836 0xc015043100}
2023-08-22 10:39:38.535 UTC [INFO]  steampipe-plugin-terraform.plugin: [INFO]  1692700778380: found pending item [terraform-1692700646836] - subscribing to its data (terraform-1692700778380)
2023-08-22 10:39:38.535 UTC [INFO]  steampipe-plugin-terraform.plugin: [INFO]  1692700778380: stream all data already cached

```

**Expected behaviour**

**Additional context**

**Comment by bigdatasourav** [comment]
@rajlearner17 @khushboo9024, The above issue is resolved in the latest Steampipe SDK v5.6.0-rc.21, but this SDK version has an issue `Error: assignment to entry in nil map` when custom structure passed in StreamListItem in ParentHydrate; we can proceed to release the plugin with updated SDK after the issue is fixed.

I have updated the plugin code with the above SDK version and bypassed the custom structure in ParentHydrate. Could you please try out [this](https://github.com/turbot/steampipe-plugin-terraform/tree/update-sdk) branch and provide your feedback?

**Comment by rajlearner17** [comment]
@bigdatasourav I confirm this new branch works perfectly fine. Thanks
